llvm/lib/Transforms/IPO/MemProfContextDisambiguation.cpp

```diff
@@ -1398,31 +1408,37 @@ void CallsiteContextGraph<DerivedCCG, FuncTy, CallTy>::
     // duplicated context ids. We have to recompute as we might have overlap
     // overlap between the saved context ids for different last nodes, and
     // removed them already during the post order traversal.
```

Comment:
Does this comment need to be updated? I guess "recompute" doesn't refer to the `getContextIds()` call that is now elided.

Also can L1402 `ContextNode *FirstNode = ...` now be moved closer to where it is used?

Reply:
> 

> Does this comment need to be updated? I guess "recompute" doesn't refer to the `getContextIds()` call that is now elided.

Right on the latter - it doesn't refer to getContextIds but rather to the recomputation that remains here.
> 
> Also can L1402 `ContextNode *FirstNode = ...` now be moved closer to where it is used?

Done